topics/statistics/tutorials/RNN/tutorial.md

```diff
@@ -0,0 +1,365 @@
+>        - *"Choose how to select data by column:"*: `All columns`
+>    - Click *"Execute"*
+>
+>
+{: .hands_on}
+
+The training step generates 3 datasets. 1) accuracy of the trained model, 2) the trained model, downloadable as a zip file, and 3) the trained 
+model weights, downloadable as an hdf5 file. These files are needed for prediction in the next step. 
+
+### **Model Prediction**
+
+> ### {% icon hands_on %} Hands-on: Testing the model
+>
+> - {% tool [Model Prediction](toolshed.g2.bx.psu.edu/repos/bgruening/model_prediction/model_prediction/1.0.8.2) %}
+>    - *"Choose the dataset containing pipeline/estimator object"* : Select the trained model from the previous step. 
+>    - *"Choose the dataset containing weights for the estimator above"* : Select the trained model weights from the previous step. 
+>    - *"Select invocation method"*: `predict`
+>    - *"Select input data type for prediction"*: `tabular data`
+>    - *"Training samples dataset"*: Select `X_test` dataset 
+>    - *"Choose how to select data by column:"*: `All columns`
+>    - Click *"Execute"*
+>
+{: .hands_on}
+
+The prediction step generates 1 dataset. It's a file that has predictions (1 or 0 for positive or negative movie reviews) for every review in 
+the test dataset.
+
+### **Machine Learning Visualization Extension**
+
+> ### {% icon hands_on %} Hands-on: Creating the confusion matrix
+>
+> - {% tool [Machine Learning Visualization Extension](toolshed.g2.bx.psu.edu/repos/bgruening/ml_visualization_ex/ml_visualization_ex/1.0.8.2) %}
+>    - *"Select a plotting type"*: `Confusion matrix for classes`
+>    - *"Select dataset containing the true labels"*": `y_test` 
+>    - *"Choose how to select data by column:"*: `All columns`
+>    - *"Select dataset containing the predicted labels"*": Select `Model Prediction` from the previous step 
+>    - *"Does the dataset contain header:"*: `Yes`
+>    - Click *"Execute"*
+>
+{: .hands_on}
+
+**Confusion Matrix** is a table that describes the performance of a classification model. It lists the number of positive and negative examples 
+that were correctly classified by the model, True positives (TP) and true negatives (TN), respectively. It also lists the number of examples that 
+were classified as positive that were actually negative (False positive, FP, or Type I error), and the number of examples that were classified 
+as negative that were actually positive (False negative, FN, or Type 2 error). Given the confusion matrix, we can calculate **precision** and 
+**recall** {% cite TatbulEtAl  %}. Precision is the fraction of predicted positives that are true positives (Precision = TP / (TP + FP)). Recall 
+is the fraction of true positives that are predicted (Recall = TP / (TP + FN)). One way to describe the confusion matrix with just one value is 
+to use the **F score**, which is the harmonic mean of precision and recall (F = (2 * precision * recall) / (precision + recall)).
```

Comment:
If you want to represent mathematical things you can do something like this, I think. (please test locally how it renders.)

```suggestion
to use the **F score**, which is the harmonic mean of precision and recall 

$$ F = \frac{2 * \text{precision * recall}}{\text{precision + recall}} $$
```

https://training.galaxyproject.org/training-material/topics/contributing/tutorials/create-new-tutorial-content/tutorial.html#writing-mathematical-expressions

Reply:
Thanks. The formula renders fine locally.